### PR TITLE
EP-6 EP-55 Transfer receive controller

### PR DESCRIPTION
**Pull request by ghiobi**
- implement receive controller logic
- deposit transfer
- receive transfer test

so far a user can receive a transfer as long as the transfer token's status is pending . Once a transfer is received the transfer will be complete.

*Please review https://github.com/Project-Epsilon/server/pull/46 first*

### CODE REVIEW

File: app/Http/Controllers/Transfer/User/ReceiveController.php [hunk 1]
```diff
@@ -2,19 +2,62 @@
...
+            'received_at' => Carbon::now()
+        ]);
+
+        return fractal()
+            ->item($transfer, new TransferTransformer())
+            ->toArray();
+    }
+
+    /**
+     * Returns the errors response.
```
Comment (ajpower) [inline app/Http/Controllers/Transfer/User/ReceiveController.php]: typo, should be error's

File: app/Http/Controllers/Transfer/User/ReceiveController.php [hunk 2]
```diff
@@ -2,19 +2,62 @@
...
+     *
+     * @param $message
+     * @return JsonErrorResponse
+     */
+    protected function sendErrorResponse($message)
+    {
+        return new JsonErrorResponse($message? : 'There was any error processing retrieval of transfer.');
```
Comment (ajpower) [inline app/Http/Controllers/Transfer/User/ReceiveController.php]: typo, should be *an

File: app/Http/Controllers/Transfer/User/SendController.php [hunk 3]
```diff
@@ -2,21 +2,107 @@
...
+
+        $user = $request->user();
+
+        $wallet = Wallet::find($request->wallet_id);
+        if (! $wallet || $wallet->user_id != $user->id) {
+            return $this->sendErrorResponse('Wallet does not exists.');
+        }
+
+        $integer = $wallet->currency->toInteger($request->amount);
+        if (((int) $integer) - $integer < 0){
```
Comment (ajpower) [inline app/Http/Controllers/Transfer/User/SendController.php]: Why not just check the length of the integer when converting to string?

File: tests/feature/TransferTest.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+     * @return @void
+     */
+    public function testReceiveTransfer()
+    {
+        $this->seed();
+
+        $transfer = Transfer::create([
+            'sender_wallet_id' => 1,
+            'amount' => '1000', //10 canadian dollars
+            'status' => 'pending',
```
Comment (JTalarico) [inline tests/feature/TransferTest.php]: No inline comments. We should mention somewhere in the docs that amount should be 10x as much.

File: app/Http/Controllers/Transfer/User/SendController.php (hunk 3 above)
Comment (ghiobi) [inline app/Http/Controllers/Transfer/User/SendController.php]: Would you like to suggest the algorithm?

File: app/Http/Controllers/Transfer/User/ReceiveController.php (hunk 2 above)
Comment (ghiobi) [inline app/Http/Controllers/Transfer/User/ReceiveController.php]: In this example, would you need to set the function as `getTheDefaultDriver()`?

https://github.com/illuminate/auth/blob/master/AuthManager.php#L183

File: app/Http/Controllers/Transfer/User/SendController.php (hunk 3 above)
Comment (JTalarico) [inline app/Http/Controllers/Transfer/User/SendController.php]: I think laurendy's way is right. The alternative is:  strlen(substr(strrchr($str, "."), 2)); which is more complicated.

**Comment by ghiobi** [comment]
**Please review #46 first** and squash please.

File: app/Http/Controllers/Transfer/User/SendController.php (hunk 3 above)
Comment (JTalarico) [inline app/Http/Controllers/Transfer/User/SendController.php]: Although I dont understand the < 0

File: tests/feature/TransferTest.php (hunk 4 above)
Comment (ghiobi) [inline tests/feature/TransferTest.php]: There should be some inline comments if the function is complex, or a variable is not obvious.

**Pull request merged**